src/games/strategy/triplea/ui/MapData.java

```diff
@@ -700,6 +701,50 @@ private Rectangle getBoundingRectWithTranslate(final List<Polygon> polys, final
   }
 
   /**
+   * Find the best rectangle inside the territory to place the name in. Finds the rectangle
+   * that can fit the name, that is the closest to the vertical center, and has a large width at
+   * that location. If there isn't any rectangles that can fit the name then default back to the
+   * bounding rectangle.
+   */
+  public Rectangle getBestTerritoryNameRect(final Territory t, final FontMetrics fm) {
```

Comment:
Because the purpose of the method is to try to better visually place the territory names. Hard to write a test to check how well a name is visually placed. Imagine even trying to do TDD for this. What tests would you write to prove the method is working as intended? Besides a basic test that shows you actually return a rectangle there isn't much.